### PR TITLE
#249 Support remote urls for static components

### PR DESCRIPTION
**Pull request by edaena**
Add support to download a remote resource manifest for a static component.

### CODE REVIEW

File: core/component.go [hunk 1]
```diff
@@ -208,6 +241,8 @@ func (c *Component) InstallComponent(componentPath string) (err error) {
 		if err = CloneRepo(c.Source, c.Version, subcomponentPath, c.Branch); err != nil {
 			return err
 		}
+	} else if c.ComponentType == "static" || c.Method == "remote-url" {
```
Comment (evanlouie) [inline core/component.go]: Should the logic operator be a `&&`?

File: core/component.go [hunk 2]
```diff
@@ -494,3 +529,33 @@ func (c *Component) GetAccessTokens() (tokens map[string]string, err error) {
...
+	}
+
+	return path.Join(cmdDir, dirPath), nil
+}
+
+// IsValidRemoteComponentConfig checks if the given component configuration is valid for a remote component
+func IsValidRemoteComponentConfig(c Component) (bool){
+	return (
+		(c.ComponentType == "static" &&
+		 c.Method != "remote-url")) &&
```
Comment (evanlouie) [inline core/component.go]: Should this be `c.Method == "remote-url"`?

File: core/component.go [hunk 3]
```diff
@@ -494,3 +536,46 @@ func (c *Component) GetAccessTokens() (tokens map[string]string, err error) {
...
+	}
+
+	return path.Join(cmdDir, dirPath), nil
+}
+
+// IsValidRemoteComponentConfig checks if the given component configuration is valid for a remote component
+func IsValidRemoteComponentConfig(c Component) (bool){
+	return (
+		(c.ComponentType == "static" &&
+		 c.Method == "remote-url")) &&
```
Comment (evanlouie) [inline core/component.go]: I'm starting to think we should call `remote-url` something else. Seeing as `method: git` is technically a _remote url_, we should choose something more concise to what is actually being used. Maybe something like `method: http`? thoughts?

**Comment by edaena** [comment]
@evanlouie `http` sounds like a good option, the other possible alternative is `remote` instead of `remote-url` since it's the opposite to `local`, but as you brought up earlier, git and helm are also remote-urls.
I'm leaning toward `http`

**Comment by evanlouie** [comment]
Lets go with `http` for now. Seems like the most accurate name for what its doing.

File: testdata/generate-remote-static/component.yaml [hunk 4]
```diff
@@ -4,6 +4,6 @@ path: ./manifests
 subcomponents:
   - name: "keyvault-sub"
     source: https://raw.githubusercontent.com/Azure/kubernetes-keyvault-flexvol/master/deployment/kv-flexvol-installer.yaml
-    method: remote-url
+    method: http
     type: static
     path: "./tmp/keyvault-sub"
```
Comment (evanlouie) [inline testdata/generate-remote-static/component.yaml]: For future reference, please setup your editor to ensure newlines on end of files (makes files more grep friendly). This is the `files.insertFinalNewline` option in vscode.

**Comment by evanlouie** [comment]
closes #249

**Pull request merged**